blockstore/badger/blockstore.go

```diff
@@ -683,13 +588,14 @@ func (b *Blockstore) Has(ctx context.Context, cid cid.Cid) (bool, error) {
 		defer KeyPool.Put(k)
 	}
 
-	err := b.db.View(func(txn *badger.Txn) error {
+	err := b.db.View(func(txn badger.Txn) error {
 		_, err := txn.Get(k)
 		return err
 	})
 
+	errKeyNotFound := b.db.GetErrKeyNotFound()
```

Comment:
This isn't great. Other options:

1. Re-write errors inside these functions. We'd need to be careful to make sure we don't miss any cases...
2. Horrible but... we _can_ `badger2.ErrKeyNotFound = badger4.ErrKeyNotFound`. Really, it's not _that_ horrible (unless someone is stashing these errors early and using those stashed version for comparison).
3. Change the APIs so we don't need to check against special errors?
4. Provide `IsKeyNotFound` etc. helpers that abstract over different badger versions? Also kind of nasty.

I can't think of a great solution, tbh.

Reply:
I looked into `badger2.ErrKeyNotFound = badger4.ErrKeyNotFound` and the current implementation feels like the most maintainable solution option to me